### PR TITLE
feat(gopls): attach previous lsp instance for GOMODCACHE

### PR DESCRIPTION
**Pull request by icholy**
Issue: https://github.com/neovim/nvim-lspconfig/issues/804
Reboots: https://github.com/neovim/nvim-lspconfig/pull/879

### CODE REVIEW

File: lua/lspconfig/server_configurations/gopls.lua [hunk 1]
```diff
@@ -1,11 +1,20 @@
 local util = require 'lspconfig.util'
 
+local last_root = nil
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/gopls.lua:3]: no need

File: lua/lspconfig/server_configurations/gopls.lua [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
+local mod_cache = vim.trim(vim.fn.system 'go env GOMODCACHE')
+
 return {
   default_config = {
     cmd = { 'gopls' },
     filetypes = { 'go', 'gomod', 'gowork', 'gotmpl' },
     root_dir = function(fname)
-      return util.root_pattern('go.work', 'go.mod', '.git')(fname)
+      -- see: https://github.com/neovim/nvim-lspconfig/issues/804
+      local fpath = vim.fn.expand(fname, ':p')
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/gopls.lua:12]:
```suggestion
      if fname:sub(1, #mod_cache) == mod_cache then
         local client = vim.lsp.get_active_clients({ name = 'gopls' })
         if client then
            return client.config.root_dir
         end
     end
     return util.root_pattern('go.work', 'go.mod', '.git')(fname)
```

File: lua/lspconfig/server_configurations/gopls.lua [hunk 3]
```diff
@@ -1,11 +1,20 @@
...
+
 return {
   default_config = {
     cmd = { 'gopls' },
     filetypes = { 'go', 'gomod', 'gowork', 'gotmpl' },
     root_dir = function(fname)
-      return util.root_pattern('go.work', 'go.mod', '.git')(fname)
+      -- see: https://github.com/neovim/nvim-lspconfig/issues/804
+      if fname:sub(1, #mod_cache) == mod_cache then
+         local client = vim.lsp.get_active_clients({ name = 'gopls' })
```
Comment (icholy) [inline lua/lspconfig/server_configurations/gopls.lua:12]: @glepnir won't this return a table?

Comment (icholy) [reply lua/lspconfig/server_configurations/gopls.lua]: done

**Comment by icholy** [comment]
@glepnir ready for review again

**Pull request merged**